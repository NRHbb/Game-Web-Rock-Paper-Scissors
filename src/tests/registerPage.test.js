/** @jest-environment jsdom */
import React from 'react'
import { render } from './testUtils.js'
import RegisterPage from '../pages/register/index.jsx'
import { useRouter } from 'next/router'
import { screen } from '@testing-library/dom'

jest.mock('next/router', () => ({
  useRouter: jest.fn()
}))

const mockRouter = {
  route: '/mock-route',
  basePath: '',
  pathname: '/mock-route',
  query: {},
  asPath: '/mock-route'
}

useRouter.mockReturnValue(mockRouter)

describe('render test', () => {
  it('renders register page without crashing', () => {
    render(<RegisterPage />)
  })
})

describe('form test', () => {
  it('form contains email input', () => {
    render(<RegisterPage />)
    const emailInput = screen.queryByLabelText('Email')
    expect(emailInput).toBeInTheDocument
  })
  it('form contains password input', () => {
    render(<RegisterPage />)
    const passwordInput = screen.queryByLabelText('Password')
    expect(passwordInput).toBeInTheDocument
  })
})
