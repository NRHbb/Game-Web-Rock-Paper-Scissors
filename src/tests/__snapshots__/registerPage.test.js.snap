// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render test renders register page without crashing 1`] = `
<Provider
  store={
    {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
